lms/djangoapps/bulk_email/tests/test_email.py

```diff
@@ -178,6 +180,102 @@ def test_send_to_all(self):
+        temp_ids = self._get_temp_query_ids()
+        self._save_query(temp_ids)
+        saved_queries = self._get_saved_queries()
+        self.assertEquals(len(saved_queries), 1)
+        return saved_queries[0]['group']
+
+    def test_send_to_query(self):
+        """
+        Make sure email send to query goes there.
+        """
+        query_id = self._setup_query()
+        StudentModule.objects.create(student=self.students[0],
+                                     course_id=self.course.id,
+                                     module_state_key=self.problem,
+                                     )
+        test_email = {
+            'action': 'Send email',
+            'send_to': query_id,
+            'subject': 'test subject for query',
+            'message': 'test message for all'
+        }
+        response = self.client.post(self.send_mail_url, test_email)
+        self.assertEquals(json.loads(response.content), self.success_content)
+
+        self.assertEquals(len(mail.outbox), 1)
+        self.assertEquals(mail.outbox[0].to[0], self.students[0].email)
+
+        students_left = StudentsForQuery.objects.all()
+        queries_left = TemporaryQuery.objects.all()
+        self.assertEquals(len(students_left), 0)
+        self.assertEquals(len(queries_left), 1) # should just have the one created from _setup_query
+
+    def test_send_to_query_no_results(self):
+        """
+        Make sure email send to query with no results handles successfully.
+        """
+        query_id = self._setup_query()
+        test_email = {
+            'action': 'Send email',
+            'send_to': query_id,
+            'subject': 'test subject for query',
+            'message': 'test message for all'
+        }
+        response = self.client.post(self.send_mail_url, test_email)
+        self.assertEquals(json.loads(response.content), self.success_content)
+        self.assertEquals(len(mail.outbox), 0)
+
+        queries_left = TemporaryQuery.objects.all()
+        self.assertEquals(len(queries_left), 1) # should just have the one created from _setup_query
+
+    def _make_query(self, query_type, joining="OR"):
+        """ Issues a query to the backend """
+        url = reverse('get_single_query', kwargs={'course_id': self.course.id})
+        function_args = [joining, 'Problem', self.problem.block_type, self.problem.block_id]
+        single_args = {
+            'filter': query_type,
+            "entityName": 'Introduction',
+        }
+        encoded_args = '/'.join(function_args)
+        query_url = '/'.join([url, encoded_args])
+        _unused_response = self.client.get(query_url, single_args)
+
+    def _get_temp_query_ids(self):
+        """ Returns all the current queries """
+        get_temp_url = reverse('get_temp_queries', kwargs={'course_id': self.course.id})
+        temp_response = self.client.get(get_temp_url)
+        queries = json.loads(temp_response.content)['queries']
+        if len(queries) > 0:
```

Comment:
Good catch, I actually just took this code straight from Jim: https://github.com/Stanford-Online/edx-platform/blob/master/lms/djangoapps/instructor/tests/test_api.py#L208